HIPs/firefly.md

```diff
@@ -0,0 +1,378 @@
+4. The Exporter signs the BOL and forwards it to the Importer, who will present it to the ship captain in order to take delivery of the cargo
+5. The ship’s captain verifies the authenticity of the BOL document, by verifying the signatures of the Carrier and the Exporter as well as the entitlement of the Importer to the cargo, and releases the cargo to the Importer
+6. The Importer notifies the Importer’s Bank of successful delivery of the cargo
+7. The Exporter gets paid by the Importer’s bank
+
+## Sequence Diagram
+
+![Sequence Diagram](../assets/sequence-diagram.svg "Sequence Diagram")
+
+The following should be clear from the diagram:
+
+- Blockchain is the trust foundation of the MPS
+- Blockchain takes care of the critical aspects of the multi-party business transaction:
+  - Identity of the document signers
+  - Documents can’t be tempered with after sharing (commitment on chain)
+  - The same BOL document can’t be sold more than once (double spend prevention with token representation)
+  - Entitlement verification (current owner of the non-fungible token)
+- Majority of the interactions in the system happen off-chain, with secure document transfers and encrypted private messages
+
+NOTE: at a high level you can choose whether to use a token to represent the off-chain message/doc or not, it's not always a requirement for the asset to be globally unique. In certain use cases, the requirement can be simply that the sender and receiver both got cryptographic proof of the fact of the exchange (by both sides providing signatures over the hash of the content). In these cases the pinning tx can simply record the hashes (and can be submitted from a random signing account).
+
+But if double spend protection is needed with tying the asset to a token that must be visible to the whole network, one can add protection of privacy leakage by submitting fake txs as noise. Since the input of the txs are only hashes, external parties can't tell if they were for real message/doc exchanges or not.
+
+## FireFly Details and Internals
+
+### Pluggable Architecture
+
+![Plugin Architecture](../assets/plugin-architecture.svg "Plugin Architecture")
+
+The plugins comprise of a thin layer of Go code, that is embedded into the FireFly Core runtime (built statically today, but extensible to dynamic plugins as the Go plugin framework matures). Then a remote agent can be written in any language, for heavy lifting required in-between the FireFly core runtime and the underlying component. This remote agent layer is optional, but encouraged for any non-trivial REST/WebSocket interactions and scrutiny is placed on the embedded Go code to ensure the layer does not compromise the HA model of the core runtime.
+
+### Scalability / High Availability
+
+#### Active/Active HA Model
+
+FireFly Core is designed to be scaled horizontally, in an active/active HA design.
+
+![HA / Scale Model](../assets/ha-scale-model.svg "HA / Scale Mode")
```

Comment:
It will help if this diagram also has node representation alongside. i.e. When we say core is scaled, does it mean only the core component of FireFly is scaled (deploy across regions in cloud) or is it that the entire FireFly node instance scaled up?